### PR TITLE
Refactoring List functions in several packages

### PR DESCRIPTION
**Pull request by ElaiShalevRH**
Refactoring List functions  to take an optional ListOptions argument

### CODE REVIEW

File: pkg/mco/mcplist.go [hunk 1]
```diff
@@ -12,10 +12,19 @@ import (
...
 
-	mcpList, err := apiClient.MachineConfigPools().List(context.Background(), listOptions)
+	if len(listOptions) == 1 {
+		passedOptions = listOptions[0]
+	} else if len(listOptions) > 1 {
+
+		return nil, fmt.Errorf("error: more than one ListOptions was passed")
+	}
+
+	glog.V(100).Infof("Listing all MCP resources with the options %v", passedOptions)
```
Comment (achuzhoy) [inline pkg/mco/mcplist.go:25]: Do we want to show this if the passedOptions is empty ?

File: pkg/deployment/list.go [hunk 2]
```diff
@@ -10,16 +10,25 @@ import (
...
 // List returns deployment inventory in the given namespace.
-func List(apiClient *clients.Settings, nsname string, options metaV1.ListOptions) ([]*Builder, error) {
-	glog.V(100).Infof("Listing deployments in the namespace %s with the options %v", nsname, options)
+func List(apiClient *clients.Settings, nsname string, options ...metaV1.ListOptions) ([]*Builder, error) {
+	passedOptions := metaV1.ListOptions{}
+
+	if len(options) == 1 {
+		passedOptions = options[0]
+	} else if len(options) > 1 {
+
```
Comment (achuzhoy) [inline pkg/deployment/list.go:19]: empty space after every "else if" in this MR

File: pkg/mco/mcplist.go [hunk 3]
```diff
@@ -12,10 +12,19 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 )
 
-func ListMCP(apiClient *clients.Settings, listOptions metav1.ListOptions) ([]*MCPBuilder, error) {
-	glog.V(100).Infof("Listing all MCP resources with the options %v", listOptions)
+func ListMCP(apiClient *clients.Settings, listOptions ...metav1.ListOptions) ([]*MCPBuilder, error) {
```
Comment (trewest) [inline pkg/mco/mcplist.go:15]: Can you change the parameter name to options so it is consistent with the other list functions?

File: pkg/olm/installplanlist.go [hunk 4]
```diff
@@ -10,14 +10,24 @@ import (
...
-	installPlanList, err := apiClient.InstallPlans(nsname).List(context.Background(), v1.ListOptions{})
+	passedOptions := v1.ListOptions{}
+
+	if len(options) == 1 {
+		passedOptions = options[0]
+	} else if len(options) > 1 {
+
+		return nil, fmt.Errorf("error: more than one ListOptions was passed")
+	}
+
```
Comment (trewest) [inline pkg/olm/installplanlist.go:29]: Can you add a glog message here

File: pkg/nto/performanceprofilelist.go [hunk 5]
```diff
@@ -2,18 +2,29 @@ package nto //nolint:misspell
...
 
+	passedOptions := goclient.ListOptions{}
+
+	if len(options) == 1 {
+		passedOptions = options[0]
+	} else if len(options) > 1 {
+
+		return nil, fmt.Errorf("error: more than one ListOptions was passed")
+	}
+
```
Comment (trewest) [inline pkg/nto/performanceprofilelist.go:25]: Please move the glog message here

**Comment by trewest** [comment]
@ElaiShalevRH Can you also update namespace, pod, bmh, clusteroperator, and the machine pkgs?

File: pkg/mco/mcplist.go (hunk 1 above)
Comment (ElaiShalevRH) [reply pkg/mco/mcplist.go:25]: No, it assumes the variable is populated. I'll condition the glog messages accordingly

File: pkg/bmh/list.go [hunk 6]
```diff
@@ -19,17 +19,31 @@ const (
...
 	if nsname == "" {
 		glog.V(100).Infof("bareMetalHost 'nsname' parameter can not be empty")
 
 		return nil, fmt.Errorf("failed to list bareMetalHosts, 'nsname' parameter is empty")
 	}
 
+	logMessage := fmt.Sprintf("Listing bareMetalHosts in the namespace %s", nsname)
+	passedOptions := goclient.ListOptions{}
+
+	if len(options) == 1 {
```
Comment (trewest) [inline pkg/bmh/list.go:32]: Hey @ElaiShalevRH , can you rewrite the if blocks as follows:
```
logMessage := fmt.Sprintf("Listing bareMetalHosts in the namespace %s", nsname)
passedOptions := goclient.ListOptions{}

if len(options) > 1 {
    glog.V(100).Infof("'options' parameter must be empty or single-valued")

    return nil, fmt.Errorf("error: more than one ListOptions was passed")
}

if len(options) == 1 {
    passedOptions = options[0]
    logMessage += fmt.Sprintf(" with the options %v", passedOptions)
}
```
I think it is a little bit easier to read

**Review by trewest — CHANGES_REQUESTED** [review]
One change

**Review by trewest — APPROVED** [review]
/lgtm

**Comment by achuzhoy** [comment]
@ElaiShalevRH  could you please resolve all the resolved conversations?

**Review by achuzhoy — APPROVED** [review]

**Pull request merged**